tests/kernels/test_causal_conv1d.py

```diff
@@ -55,131 +54,138 @@ def causal_conv1d_ref(
+                                 width - 1,
+                                 device=device,
+                                 dtype=itype)
+    x_ref = x.clone()
+    weight_ref = weight.clone()
+    bias_ref = bias.clone() if bias is not None else None
+    initial_states_ref = initial_states.clone(
     ) if initial_states is not None else None
     activation = None if not silu_activation else "silu"
-    out, final_states = causal_conv1d_fn(
-        x,
-        weight,
-        bias,
-        initial_states=initial_states,
-        return_final_states=return_final_states,
-        activation=activation)
+    out, final_states = causal_conv1d_fn(x,
+                                         weight,
+                                         bias,
+                                         activation=activation,
+                                         conv_states=initial_states,
+                                         has_initial_state=torch.ones(
+                                             batch,
+                                             dtype=torch.bool,
+                                             device=x.device))
     out_ref, final_states_ref = causal_conv1d_ref(
         x_ref,
         weight_ref,
         bias_ref,
         initial_states=initial_states_ref,
-        return_final_states=return_final_states,
+        return_final_states=True,
         activation=activation)
-    if return_final_states:
-        assert final_states is not None and final_states_ref is not None
-        assert torch.allclose(final_states,
-                              final_states_ref,
-                              rtol=rtol,
-                              atol=atol)
+    assert final_states is not None and final_states_ref is not None
+    assert torch.allclose(final_states, final_states_ref, rtol=rtol, atol=atol)
 
     assert torch.allclose(out, out_ref, rtol=rtol, atol=atol)
 
-    if return_final_states:
-        out += F.sigmoid(final_states).sum(dim=-1, keepdim=True)
-        out_ref += F.sigmoid(final_states_ref).sum(dim=-1, keepdim=True)
-
 
 @pytest.mark.parametrize("itype", [torch.bfloat16])
 @pytest.mark.parametrize("silu_activation", [False, True])
 @pytest.mark.parametrize("has_bias", [False, True])
+@pytest.mark.parametrize("has_cache_seqlens", [False, True])
+@pytest.mark.parametrize("seqlen", [1, 4, 5])
 @pytest.mark.parametrize("width", [2, 3, 4])
 @pytest.mark.parametrize("dim", [2048, 2048 + 16, 4096])
-@pytest.mark.parametrize("batch", [1, 2])
-def test_causal_conv1d_update(batch, dim, width, has_bias, silu_activation,
-                              itype):
+def test_causal_conv1d_update(dim, width, seqlen, has_cache_seqlens, has_bias,
+                              silu_activation, itype):
     device = "cuda"
     rtol, atol = (3e-4, 1e-3) if itype == torch.float32 else (3e-3, 5e-3)
     if itype == torch.bfloat16:
         rtol, atol = 1e-2, 5e-2
     # set seed
+    torch.random.manual_seed(0)
```

Comment:
nit: it looks like the `seed_everything` call a few lines down already takes care of this?

Reply:
Addressed

Reply:
Addressed